### ISSUE TITLE
py> operator doesn't work with continuum/miniconda3 Docker image

### ISSUE THREAD

**Issue by chezou** [treasure-data/digdag#887]
While py> operator with Python official docker image works fine, the digdag workload fails with error using [anaconda Docker image](https://hub.docker.com/r/continuumio/miniconda3/). Anaconda is a popular Python distribution especially for data science and machine learning. It would be nice if anaconda image works fine for ML workload on digdag server.

```py-test.dig
+run:
  docker:
    image: 'continuumio/miniconda3'
  py>: simple_task.run
```

```simple_task.py
def run():
    print("Hello py> operator!")

if __name__ == '__main__':
    run()
```

Error logs on digdag server:

```
$ digdag push pyop-test && digdag start pyop-test py-test --session now
``

```error.log
2018-09-28 14:48:15 +0900 [INFO] (XNIO-1 task-14): Starting a new session project id=1 workflow name=py-test session_time=2018-09-28T05:48:15+00:00
2018-09-28 14:48:20 +0900 [INFO] (0063@[0:pyop-test]+py-test+run): py>: simple_task.run
[FATAL tini (6)] exec python failed: No such file or directory
2018-09-28 14:48:22 +0900 [ERROR] (0063@[0:pyop-test]+py-test+run): Task failed with unexpected error: Python command failed with code 127
java.lang.RuntimeException: Python command failed with code 127
	at io.digdag.standards.operator.PyOperatorFactory$PyOperator.runCode(PyOperatorFactory.java:153)
	at io.digdag.standards.operator.PyOperatorFactory$PyOperator.runTask(PyOperatorFactory.java:91)
	at io.digdag.util.BaseOperator.run(BaseOperator.java:35)
	at io.digdag.core.agent.OperatorManager.callExecutor(OperatorManager.java:312)
	at io.digdag.core.agent.OperatorManager.runWithWorkspace(OperatorManager.java:254)
	at io.digdag.core.agent.OperatorManager.lambda$runWithHeartbeat$2(OperatorManager.java:137)
	at io.digdag.core.agent.ExtractArchiveWorkspaceManager.withExtractedArchive(ExtractArchiveWorkspaceManager.java:77)
	at io.digdag.core.agent.OperatorManager.runWithHeartbeat(OperatorManager.java:135)
	at io.digdag.core.agent.OperatorManager.run(OperatorManager.java:119)
	at io.digdag.core.agent.MultiThreadAgent.lambda$null$0(MultiThreadAgent.java:127)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
2018-09-28 14:48:27 +0900 [INFO] (0063@[0:pyop-test]+py-test^failure-alert): type: notify
```

If I use [python 3.7.0 official image](https://hub.docker.com/_/python/), digdag workload works fine.

```python370.log
2018-09-28 14:48:27 +0900 [INFO] (0063@[0:pyop-test]+py-test^failure-alert): type: notify
2018-09-28 15:10:31 +0900 [INFO] (XNIO-1 task-22): Starting a new session project id=1 workflow name=py-test session_time=2018-09-28T06:10:31+00:00
2018-09-28 15:10:37 +0900 [INFO] (0065@[0:pyop-test]+py-test+run): py>: simple_task.run
Hello py> operator!
<stdin>:4: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
<stdin>:111: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
```

Files are put as follows:

```
$ tree .
.
├── py-test.dig
└── simple_task.py

0 directories, 2 files
```

You can see detailed contents at gist:
https://gist.github.com/chezou/bdf1729bcf4254dcffcf7558b53a42e5

Version info:

```
$ digdag --version
0.9.30
$ java --version
java 10.0.2 2018-07-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.2+13)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)
$ docker --version
Docker version 18.06.1-ce, build e68fc7a
```

**Comment by hiroyuki-sato** [comment]
Hello, @chezou 

Could you try the following?

* Set the following PATH environment variable on digdag server.
* Start Digdag server 
* Run `digdag push pyop-test && digdag start pyop-test py-test --session now`

Digdag use server PATH envirionment variable on Docker image.
And your environment variable may not set `/opt/conda/bin` in Digdag server.
(This path the python location in `continuumio/miniconda3` image).

```
export PATH="/opt/conda/bin:$PATH"
digdag server -m
```

```
digdag push pyop-test && digdag start pyop-test py-test --session now
```

### Note

This issue also related #609 #611 and #680

**Comment by chezou** [comment]
@hiroyuki-sato Thanks for your immediate response! I confirmed your work around can avoid the issue.

Just out of curiosity, is it standard way to set `PATH` environment for specific docker image before starting server? I worries this prerequisite may cause annoying negotiation between administrator of digdag server and users.

**Comment by hiroyuki-sato** [comment]
@chezou 
It's just the workaround. I'm not sure detail but #835 will introduce new CommandExecutor.
It may solve a part of your issue.
The Developer will talk about Digdag in the upcoming event  [TDTech](https://techplay.jp/event/694468) in Japan. I'll tell him about this behavior.

@muga What do you think? Does CommandExecutor v2 solve this issue?

**Comment by muga** [comment]
@chezou @hiroyuki-sato thank you for reporting. At least the issue could be reproduced on v0_10 branch. I will take a look to find a way to avoid this issue.

**Comment by muga** [comment]
Filed https://github.com/treasure-data/digdag/pull/890

**Comment by muga** [comment]
I'm closing this ticket because the issue can be fixed by Digdag v0.9.31 released recently. Please feel free to reopen it if need.
